### PR TITLE
Enable nullability in OwnerDrawPropertyBag

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in OwnerDrawPropertyBag.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4653)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs [hunk 1]
```diff
@@ -42,7 +40,7 @@ internal OwnerDrawPropertyBag()
         {
         }
 
-        public Font Font { get; set; }
+        public Font? Font { get; set; }
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:43]: wouldn't this a breaking change? is this change required because of change at line 26?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:43]: We have a `Font is null` check in `IsEmpty` method, so I assumed that this could be null.

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs [hunk 2]
```diff
@@ -25,15 +23,15 @@ protected OwnerDrawPropertyBag(SerializationInfo info, StreamingContext context)
             {
                 if (entry.Name == nameof(Font))
                 {
-                    Font = (Font)entry.Value;
+                    Font = entry.Value as Font;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:26]: Wouldn't we changing the behavior by doing this?

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:43]: Yes Font can be null. I am trying to understand why this need be declared as nullable? isn't  it default to null?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:43]: When we enable nullability, we need to specify which members can be null and which not.

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs (hunk 2 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:26]: Looking at the references, we may be safe to keep it as-is.

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs [hunk 3]
```diff
@@ -25,15 +23,15 @@ protected OwnerDrawPropertyBag(SerializationInfo info, StreamingContext context)
             {
                 if (entry.Name == nameof(Font))
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:24]:
```suggestion
                if (entry.Name == nameof(Font) && entry.Value is Font font)
                    Font = font;
```

**Review by dreddy-work — DISMISSED** [review]
Have nits otherwise LGTM.

File: src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs [hunk 4]
```diff
@@ -54,7 +52,7 @@ internal Gdi32.HFONT FontHandle
             {
                 if (_fontWrapper is null)
                 {
-                    _fontWrapper = new Control.FontHandleWrapper(Font);
+                    _fontWrapper = new Control.FontHandleWrapper(Font!);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/OwnerDrawPropertyBag.cs:55]: I understand this method has a single callsite, which does check for `null`. But maybe it is better to add an assert here (to catch possible unsuspecting callers) instead of `!`.

**Comment by RussKie** [comment]
@gpetrou a gentle nudge :)

**Comment by dreddy-work** [comment]
@gpetrou,  you may want to rebase or resolve conflict so we can merge now.

**Review by RussKie — APPROVED** [review]
